bin/bitbake-setup

```diff
@@ -236,7 +242,8 @@ def update_build(config, confdir, builddir, layerdir, d):
             layer_config[k]["git-remote"] = v["git-remote"]
     checkout_layers(layer_config, layerdir, d)
     bitbake_config = config["bitbake-config"]
-    setup_bitbake_build(bitbake_config, layerdir, builddir)
+    thisdir = os.path.dirname(os.path.abspath(config["path"]))
```

Comment:
Please run
```
$ bin/bitbake-selftest -v bb.tests.setup
```
before submitting patches, as this causes a regression in the tests :)

The above needs to say something like:
```
thisdir = os.path.dirname(config["path"] if config["type"] == 'local' else None)
```
and then setup_bitbake_build should raise an exception if THISDIR is in the config, but thisdir is None (e.g. the config is coming from http source, or the registry).